tests/fsharp/single-test.fs

```diff
@@ -71,54 +69,49 @@ let generateProps testCompilerVersion configuration =
   <Import Project=""$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(PROJECTDIRECTORY)'))"" />
 </Project>"
     template
-    |> replaceTokens "$(TESTCONFIGURATION)" configuration
-    |> replaceTokens "$(PROJECTDIRECTORY)" (Path.GetFullPath(__SOURCE_DIRECTORY__))
-    |> replaceTokens "$(TESTCOMPILERVERSION)" testCompilerVersion
+      .Replace("$(TESTCONFIGURATION)", configuration)
+      .Replace("$(PROJECTDIRECTORY)", Path.GetFullPath(__SOURCE_DIRECTORY__))
+      .Replace("$(TESTCOMPILERVERSION)", testCompilerVersion)
 
-let generateTargets =
+let targetsBody =
     let template = @"<Project>
   <Import Project=""$([MSBuild]::GetPathOfFileAbove('Directory.Build.targets', '$(PROJECTDIRECTORY)'))"" />
   <Import Project=""$(MSBuildThisFileDirectory)Directory.Overrides.targets"" Condition=""'$(OutputType)'=='Script'"" />
 </Project>"
     template
-    |> replaceTokens "$(PROJECTDIRECTORY)" (Path.GetFullPath(__SOURCE_DIRECTORY__))
+     .Replace("$(PROJECTDIRECTORY)", Path.GetFullPath(__SOURCE_DIRECTORY__))
 
-let generateOverrides =
+let overridesBody =
     let template = @"<Project>
   <Target Name=""Build"" DependsOnTargets=""RunFSharpScript"" />
   <Target Name=""Rebuild"" DependsOnTargets=""RunFSharpScript"" />
 </Project>"
     template
 
+let computeSourceItems (pc: ProjectConfiguration) addDirectory addCondition (compileItem: CompileItem) sources =
```

Comment:
@KevinRansom There's no semantic change here, just getting rid of a couple of intermediate functions